### PR TITLE
2021 Strategic Planning Document

### PR DESCRIPTION
**Pull request by kmpaul**
I have made an attempt to summarize our conclusions from the Start/Stop/Continue exercise we held over the last few weeks.  Please let me know if this looks good to you, if it is missing something, and what we could change.

Note that I have essentially included all of the "What we do Ideas" unchanged, and I have proposed a *Project Cycle* to the way we do our business to address the "How we work" comments.

Please provide reviews and feedback as you see fit.

### CODE REVIEW

**Review by halehawk — COMMENTED** [review]
I looked the whole plan, I thought the suggested "Processes" is a very good idea, though it needs more efforts from everybody, we definitely can try.
The activities that you listed, I thought you missed "Create a notebook gallery or alternative method for our team to share advanced Python codes and Pangeo codes [++]
Outreach examples that use multiple projects' code to show interoperability between Pangeo projects []
".
Do you think this one is part of other activities?

File: strategy/plan2021.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+of our activities and reduce siloing (i.e., eliminate single points
+of failure).  Almost all of the brainstorming items from this part of
+the exercise can be addressed by creating a more engaged, inclusive,
+and welcoming team environment.  This is a very hard problem to solve,
+but it is one that we should all take ownership for and strive to
+overcome.
+
+I do not suggest that this is the solution, but guided by suggestions
+that were made in the exercise, I am going to propose a solution to
+this problem that involves imposing a 4-week "Project Cycle" that looks
```
Comment (andersy005) [inline strategy/plan2021.md:45]: Looking forward to these project cycles... Great plan 👍🏽

**Review by andersy005 — APPROVED** [review]

**Review by jukent — APPROVED** [review]
Looks great! Looking forward to this cycle structure.

**Review by sherimickelson — COMMENTED** [review]
The idea of the project cycles sounds interesting, but I have a couple of concerns.  
One being that pitching a project requires you to be a project lead.  This seems like a large commitment that might be difficult for some of us that aren't supported by xdev to commit to and it may lead to only a couple members always having to take on this role.  
Second, it may be difficult for those who are not supported by xdev to know how much they can contribute towards a project within a months time when demands are changing.  Thus the timeline and commitments might inhibben some really great initiatives from even getting started. 
I understand the possible need for a rigid structure, but I'm worried that the rigidness might hinder people from contributing to different parts throughout the process.   
I think more flexible language put into it would make for a more inviting space and allow people to contribute more.

File: strategy/plan2021.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+## Activities: *What we do*
+
+Many ideas for projects were given in the *Activities* section of the
+Start/Stop/Continue exercise, and many continuing Activities were listed
+as high priority.  No Activities were lists in the Stop category, indicating
+*to me* that what we feel we need to do most is be more productive.  Hence,
+for future reference, I include all of the mentioned activities in prioritized
+order by vote count.
+
+- **[OLD]** More collaborations with the GeoCAT team like the GeoCAT Visualization Gallery Hackathon [9]
```
Comment (erogluorhan) [inline strategy/plan2021.md:104]: I am glad and feeling responsible to see such a high number of votes from the Xdev team to increase collaboration with GeoCAT. The GeoCAT team is sharing the same motivation. Methods of how to achieve these could be discussed in the upcoming project cycles of course; however, I just want to write down a few channels we can make use of currently (which @pilotchute has likely have shared in the weekly meeting):

- GeoCAT has been holding weekly hack days on every Friday (between 10am - 3pm every other Friday, between 1 pm - 3 pm rest of the every other Fridays), where we discuss/share, e.g.:
    - Newly learned technologies
    - Technical stuff that any team member is recently struggling 
    - Any issue that needs to be discussed as a team,
    - Plan of a new development approach
    - Etc.
In these events, many topics, which would be interest of both teams (such as Xarray, Dask, code coverage, code quality, CI, etc.), are being discussed and worked hands-on when needed. I am happy to have these weekly events all together with Xdev. If a weekly frequency is more than enough, we can dedicate at least one of them monthly to our collaboration.

Let us further discuss this and other possibilities.

**Review by erogluorhan — APPROVED** [review]
I missed the first two cycles; however, having these cycles as the whole team and determining the entire team's strategic plan with Start/Stop/Continue brainstorming sounds like a great inclusive idea. I am planning to attend to the rest of such cycles, time permitting.

**Comment by kmpaul** [comment]
@sherimickelson Keep in mind that these projects are for work that takes essentially only 2 weeks. That shouldn't be too much work, IMHO.

**Comment by kmpaul** [comment]
Also, I would always be involved to help plan and scope projects.

**Comment by kmpaul** [comment]
> I looked the whole plan, I thought the suggested "Processes" is a very good idea, though it needs more efforts from everybody, we definitely can try.
> The activities that you listed, I thought you missed "Create a notebook gallery or alternative method for our team to share advanced Python codes and Pangeo codes [++]
> Outreach examples that use multiple projects' code to show interoperability between Pangeo projects []
> ".
> Do you think this one is part of other activities?

@halehawk:  Yes.  I left that out because after looking more closely at it, it is already something that we are doing.  So, it's not a new activity.  There is the https://github.com/NCAR/notebook-gallery already, and we are trying to hook this into the Pangeo Gallery already. ...And frankly, I think that we should be contributing to the Pangeo Gallery more.

File: strategy/plan2021.md (hunk 2 above)
Comment (kmpaul) [reply strategy/plan2021.md:104]: While I personally do not work on Fridays, I think many Xdev members would love to be invited to those hack days.  It's a great way of concentrating our collaborative efforts!

**Comment by kmpaul** [comment]
@sherimickelson:  Do you still have concerns?  I agree that those people who are at a roughly 10-20% level in the group might find it hard to work in a cyclic nature like this.  Perhaps we can keep the rigid cycle and make involvement and level in projects more flexible?  I understand your concern about those people who are at a lower-level in the group staying up-to-date with work that others are doing, but perhaps the real problem is that people who can only give 10-20% of their time to Xdev don't have enough time to actually stay up-to-date...?

**Comment by bonnland** [comment]
> the real problem is that people who can only give 10-20% of their time to Xdev don't have enough time to actually stay up-to-date...?

Is there a way of addressing the differences in time commitment within the document itself?   I understand it's nice to have language that is maximally inclusive, i.e. everyone can participate in any role, but IMHO it's better to address the reality of different time commitments within the document.  

From my point of view, most of my work is being captured by a completely different and highly structured process (Agile Scrum and JIRA).  It feels like a huge lift to become deeply committed to more than one software development process.   It's a problem with a lot of history behind it at NCAR, and I would encourage some research into past problems and solutions around this issue.   Eric N. may be a reference here.

**Comment by halehawk** [comment]
On Fri, Jan 29, 2021 at 2:40 PM Kevin Paul <notifications@github.com> wrote:

> I looked the whole plan, I thought the suggested "Processes" is a very
> good idea, though it needs more efforts from everybody, we definitely can
> try.
> The activities that you listed, I thought you missed "Create a notebook
> gallery or alternative method for our team to share advanced Python codes
> and Pangeo codes [++]
> Outreach examples that use multiple projects' code to show
> interoperability between Pangeo projects []
> ".
> Do you think this one is part of other activities?
>
> @halehawk <https://github.com/halehawk>: Yes. I left that out because
> after looking more closely at it, it is already something that we are
> doing. So, it's not a new activity. There is the
> https://github.com/NCAR/notebook-gallery already, and we are trying to
> hook this into the Pangeo Gallery already. ...And frankly, I think that we
> should be contributing to the Pangeo Gallery more.
>
Good to know.

**Comment by kmpaul** [comment]
@bonnland @sherimickelson 

Let me first apologize for my curtness in my last comment.  I should not try to write things when I do not have time to write them properly. 

I would like to address the issue of level of commitment within the group. I realize it is hard to find a system that accommodates all levels of commitment, but it does seem that there is agreement, indicated by the Start/Stop/Continue exercise, that we should make it easier for team members to collaborate with each other and on each other's work/projects. This includes:

- making it easier to spin up on other projects,
- making meetings more about conversation and less about check ins, and
- having more frequent status checks and realignment with group priorities.

The proposed Project Cycle, as pointed out, works well for full-time Xdev members, but not for part time members.  How can we address the above points in a way that satisfies both full-time and part-time members?

**Comment by kmpaul** [comment]
(And, obviously, I can scrap the proposed Project Cycle if we develop something better.)

**Comment by kmpaul** [comment]
@bonnland By the way, when you say "in the document itself," are you referring to the Strategic Planning 2021 document I just added?  Or to the Project Specification documents referenced in the Strategic Planning 2021 document?

**Comment by bonnland** [comment]
> @bonnland By the way, when you say "in the document itself," are you referring to the Strategic Planning 2021 document I just added? Or to the Project Specification documents referenced in the Strategic Planning 2021 document?

@kmpaul  Perhaps I have tunnel vision right now from coding all day, apologies...I meant the document in this pull request.   I haven't digested the other documents yet.

**Comment by kmpaul** [comment]
@bonnland No worries. I just wanted to clarify.

**Comment by sherimickelson** [comment]
@kmpaul , I think the process is well thought out, especially if you're able to help through the planning stage. I think my  concern is the more of the timeline itself.  It's hard enough to scope a project out, but I think scoping a project to fit within a couple of weeks is difficult, especially when you don't know how many people will be able to help out when you're proposing the project.  Now this could be a learning experience in itself which could be valuable, but having such a rigid schedule might feel like we have more failures than successes.  Also, It sounds as if we would have to commit a couple of weeks before the work would start and availability might be hard to predict.  If one or people fall off the project, what happens to the timeline?  This is part of the  reason why people still have issues attached to them, but they haven't been completing them.
So I think all of the right pieces are there to try a couple rounds of this, but I think we should use language like "a proposed timeline".  I realize that under this working plan things shouldn't drag on forever, but maybe we can give ourselves an extra month in the timeline if a project needs it, especially in the beginning.  This could bring more people onto the project knowing that if something comes up, they might have an extra week to contribute.  This could also help encourage people to sign up for an additional project at the same time.

**Comment by kmpaul** [comment]
Ok. I appreciate the feedback, everyone. Considering what I've heard, I think maybe I pulled the trigger on the Project Cycle proposal a little early. I'm going to go back to the drawing board and look more closely at what was written in the Start/Stop/Continue exercise.  Perhaps we can iterate on some ideas I will try to develop in an issue before I attempt a draft of this document.  I genuinely believe we should find a solution that accommodates everyone's needs. 

Thanks, everyone!

**Pull request closed without merging**